pages/projects/projects-spacetimecube.tsx

```diff
@@ -102,6 +121,46 @@ const ProjectSpaceTimeCube: NextPage<Props> = ({
       });
     },
   );
+  const options = {
+    responsive: true,
+    plugins: {
+      legend: {
+        position: "top" as const,
+      },
+      title: {
+        display: true,
+        text: "projects in selected year",
+      },
+    },
+  };
+
+  const labels = Array.from(
+    selectedYear &&
+      projectsByYearCountry.filter((d) => d.year == selectedYear)[0]?.countries
```

Comment:
Instead of `.filter(…)[0]` you can use `.find(…)` here, which returns a single array member (or undefined) instead of an array.

Reply:
I think this code is more complicated than it needs to be as we always need to check whether `selectedYear` is defined. What about moving the entire code into a separate component (similar to `SelectedProjectDetails` and only render it conditionally (only if `selectedYear` is defined). Then within the component we can always assume that `selected Year` is defined 🙂

Reply:
Ok.